data_model/src/events/pipeline.rs

```diff
@@ -130,13 +134,24 @@ impl Event {
     /// Construct [`Event`].
     pub const fn new(entity_kind: EntityKind, status: Status, hash: Hash) -> Self {
         Event {
+            domain_id: None,
             entity_kind,
             status,
             hash,
         }
     }
 }
 
+impl GenericEvent for Event {
+    fn scoped_domain_id(&self) -> Option<<crate::prelude::Domain as crate::Identifiable>::Id> {
+        self.domain_id.clone()
+    }
+
+    fn event_type(&self) -> super::EventType {
+        EventType::Pipeline
+    }
+}
+
```

Comment:
Ok, I've got why we need `GenericEvent` implemented for all events... But it still looks like a workaround...

Reply:
The alternative way is to keep the domain id inside `Filter` and `Event` and match them in the `match()` method. But it is difficult because `Filter` doesn't know about the domain and trigger.